apps/token/src/routes/governance/propose/network-parameter/propose-network-parameter.tsx

```diff
@@ -0,0 +1,292 @@
+                  <Select
+                    data-testid="proposal-parameter-select"
+                    id="proposal-parameter-key"
+                    {...register('proposalNetworkParameterKey', {
+                      required: t('Required'),
+                    })}
+                    onChange={(e) => setSelectedNetworkParam(e.target.value)}
+                    value={selectedNetworkParam}
+                  >
+                    <option value="">{t('SelectParameter')}</option>
+                    {networkParamsData?.map(({ key }) => (
+                      <option key={key} value={key}>
+                        {key}
+                      </option>
+                    ))}
+                  </Select>
+                  {errors?.proposalNetworkParameterKey?.message && (
+                    <InputError intent="danger">
+                      {errors?.proposalNetworkParameterKey?.message}
+                    </InputError>
+                  )}
+                </FormGroup>
+
+                {selectedNetworkParam && (
+                  <div className="mt-[-10px]">
+                    <SelectedNetworkParamCurrentValue
+                      value={
+                        networkParamsData?.find(
+                          ({ key }) => key === selectedNetworkParam
+                        )?.value || ''
+                      }
+                    />
+
+                    <FormGroup
+                      label={t('NewProposedValue')}
+                      labelFor="proposal-parameter-new-value"
+                    >
+                      <TextArea
+                        data-testid="selected-proposal-param-new-value"
+                        id="proposal-parameter-new-value"
+                        {...register('proposalNetworkParameterValue', {
+                          required: t('Required'),
+                        })}
+                      />
+                      {errors?.proposalNetworkParameterValue?.message && (
+                        <InputError intent="danger">
+                          {errors?.proposalNetworkParameterValue?.message}
+                        </InputError>
+                      )}
+                    </FormGroup>
+                  </div>
+                )}
+
+                <ProposalFormVoteAndEnactmentDeadline
+                  voteRegister={register('proposalVoteDeadline', {
+                    required: t('Required'),
+                  })}
+                  voteErrorMessage={errors?.proposalVoteDeadline?.message}
+                  voteMinClose={minVoteDeadline as string}
+                  voteMaxClose={maxVoteDeadline as string}
+                  enactmentRegister={register('proposalEnactmentDeadline', {
+                    required: t('Required'),
+                  })}
+                  enactmentErrorMessage={
+                    errors?.proposalEnactmentDeadline?.message
+                  }
+                  enactmentMinClose={minEnactmentDeadline as string}
```

Comment:
chore (blocking): We should tidy up these type casts here. You are eventually calling toUpperCase on the minEnactmentDeadline which could result in errors.

Either ensure that minEnactmentDeadline is a string or allow passing undefined for these two props.